src/earn/hooks.ts

```diff
@@ -94,3 +92,60 @@ export function useDepositEntrypointInfo({
   }, [asyncExchanges.result])
   return { hasDepositToken, hasTokensOnSameNetwork, hasTokensOnOtherNetworks, canCashIn, exchanges }
 }
+
+// Used on EarnEnterAmount.tsx with on user input with a debounced callback
+export function usePrepareEnterAmountTransactionsCallback(
+  mode: Extract<EarnActiveMode, 'deposit' | 'withdraw' | 'swap-deposit'>
+) {
+  const prepareTransactions = useAsyncCallback(
+    async (args) => {
+      return mode === 'withdraw'
+        ? prepareWithdrawTransactionsWithSwap(args)
+        : prepareDepositTransactions(args)
+    },
+    {
+      onError: (err) => {
+        const error = ensureError(err)
+        Logger.error(TAG, 'usePrepareEnterAmountTransactions - Error:', error)
+      },
+    }
+  )
+
+  return {
+    prepareTransactionsResult: prepareTransactions.result,
+    refreshPreparedTransactions: prepareTransactions.execute,
+    clearPreparedTransactions: prepareTransactions.reset,
+    prepareTransactionError: prepareTransactions.error,
+    isPreparingTransactions: prepareTransactions.loading,
+  }
+}
+
+// Used on the EarnConfirmationScreen.tsx called once on load
+export function usePrepareEnterConfirmationScreenTransactions(
```

Comment:
nit: i think you mean to name it usePrepareConfirmationScreenTransactions

Reply:
Renamed to `usePrepareEarnConfirmationScreenTransactions` in d1558f3e935968571b4b2e54528fb0f1de204535.